includes/Modules/Analytics_4/Report/Custom_Dimensions_Response_Parser.php

```diff
@@ -0,0 +1,170 @@
+	 * @return string JSON encoded string of comma separated category names (or their original IDs if no name is found).
+	 */
+	public function get_post_category_names( $category_ids_string ) {
+		$category_ids = explode( ',', $category_ids_string );
+
+		// Explode converts all split values to strings. So we cast any numeric
+		// strings to `int` so that if a display name is not found for a
+		// category_id, then the original category_id int can be passed
+		// through directly in the response.
+		$category_ids = array_map(
+			function ( $id ) {
+				return is_numeric( $id ) ? (int) $id : $id;
+			},
+			$category_ids
+		);
+
+		$category_names = array();
+		foreach ( $category_ids as $category_id ) {
+			if ( ! is_numeric( $category_id ) ) {
+				$category_names[] = $category_id;
+				continue;
+			}
+
+			if ( ! isset( $this->cache_map[ Analytics_4::CUSTOM_DIMENSION_POST_CATEGORIES ][ $category_id ] ) ) {
+				$term = get_term( $category_id );
+				$this->cache_map[ Analytics_4::CUSTOM_DIMENSION_POST_CATEGORIES ][ $category_id ] = isset( $term->name ) ? $term->name : $category_id;
+			}
+
+			$category_names[] = $this->cache_map[ Analytics_4::CUSTOM_DIMENSION_POST_CATEGORIES ][ $category_id ];
+		}
+
+		return wp_json_encode( $category_names );
+	}
+
+	/**
+	 * Swaps the IDs of any custom dimensions within the response with their respective display names.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Google_Service_AnalyticsData_RunReportResponse Response with any IDs of custom dimensions converted to their display names.
+	 */
+	public function swap_custom_dimension_ids_with_names() {
+		if ( $this->response->getRowCount() === 0 ) {
+			return $this->response;
+		}
+
+		$dimension_headers = $this->response->getDimensionHeaders();
+
+		// Create a map of any custom dimension to its equivalent parsing function to avoid
+		// looping through report rows multiple times below.
+		$custom_dimension_map = array();
+		foreach ( $dimension_headers as $dimension_key => $dimension ) {
+			if ( Analytics_4::CUSTOM_EVENT_PREFIX . Analytics_4::CUSTOM_DIMENSION_POST_AUTHOR === $dimension['name'] ) {
+				$custom_dimension_map[ $dimension_key ] = array( $this, 'get_post_author_name' );
+			}
+
+			if ( Analytics_4::CUSTOM_EVENT_PREFIX . Analytics_4::CUSTOM_DIMENSION_POST_CATEGORIES === $dimension['name'] ) {
+				$custom_dimension_map[ $dimension_key ] = array( $this, 'get_post_category_names' );
+			}
+		}
+
+		if ( empty( $custom_dimension_map ) ) {
+			return $this->response;
+		}
+
+		$rows = $this->response->getRows();
+
+		foreach ( $rows as $row ) {
+			foreach ( $custom_dimension_map as $dimension_key => $callable ) {
+				$dimension_value     = $row['dimensionValues'][ $dimension_key ]->getValue();
+				$new_dimension_value = call_user_func( $callable, $dimension_value );
+				$row['dimensionValues'][ $dimension_key ]->setValue( $new_dimension_value );
+			}
+		}
+
+		$this->response->setRows( $rows );
+		return $this->response;
```

Comment:
This operation mutates the rows of the given response, which is evident in the usage but even the tests don't use the return value.

Let's remove the return to make this more explicit (if nothing is returned, the operation must mutate the given report).

```suggestion
		$this->response->setRows( $rows );
```

(also in `@return` above)